src/rust/kafka/src/lib.rs

```diff
@@ -150,6 +157,71 @@ impl<T: SerDe> Producer<T> {
     }
 }
 
+#[derive(Clone)]
+pub struct BytesProducer {
+    producer: FutureProducer,
+    topic: String,
+}
+
+impl BytesProducer {
+    pub fn new(config: ProducerConfig) -> Result<Self, ConfigurationError> {
+        Ok(Self {
+            producer: producer(
+                config.bootstrap_servers,
+                config.sasl_username,
+                config.sasl_password,
+            )?,
+            topic: config.topic,
+        })
+    }
+
+    #[tracing::instrument(err, skip(self))]
+    pub async fn send(&self, msg: Bytes) -> Result<(), ProducerError> {
+        let record: FutureRecord<[u8], [u8]> = FutureRecord::to(&self.topic).payload(&msg);
+
+        self.producer
+            .send(record, Timeout::Never)
```

Comment:
What would be a safe alternative here?

Reply:
idk, a minute?

Reply:
In this particular case, then, we'd fail to persist a retry after 1 minute. What would we do if a retry send timed out?

I think two things are important here. One, the story with respect to retries isn't done yet. I've written some of that up here, and I plan to go into more detail in an RFC soon: https://grapl-internal.slack.com/archives/C0174A8QV2S/p1657920901772039

Second, Kafka is our persistence layer. So we make sure we don't commit a consumer offset if we haven't handled the message--e.g. sent it downstream to the next topic. In light of that, I think making the producer try to send forever is the right decision, because if it _did_ time out I can't think of anything else to do... it's not as if it's somehow unblocking a connection that could otherwise be used for productive work.